Sources/SwiftInspectorCore/PropertyAnalyzer.swift

```diff
@@ -0,0 +1,248 @@
+}
+
+// MARK: - TypeLocationSyntaxVisitor
+
+private final class TypeSyntaxVisitor: SyntaxVisitor {
+
+  init(onNodeVisit: @escaping (_ info: TypeWithPropInfo) -> Void) {
+    self.onNodeVisit = onNodeVisit
+  }
+
+  override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
+    processNode(node, withName: node.identifier.text, members: node.members.members)
+    return .visitChildren
+  }
+
+  override func visit(_ node: EnumDeclSyntax) -> SyntaxVisitorContinueKind {
+    processNode(node, withName: node.identifier.text, members: node.members.members)
+    return .visitChildren
+  }
+
+  override func visit(_ node: ProtocolDeclSyntax) -> SyntaxVisitorContinueKind {
+    processNode(node, withName: node.identifier.text, members: node.members.members)
+    return .visitChildren
+  }
+
+  override func visit(_ node: StructDeclSyntax) -> SyntaxVisitorContinueKind {
+    processNode(node, withName: node.identifier.text, members: node.members.members)
+    return .visitChildren
+  }
+
+  override func visit(_ node: ExtensionDeclSyntax) -> SyntaxVisitorContinueKind {
+    if let typeIdentifier = node.extendedType.as(SimpleTypeIdentifierSyntax.self) {
+      processNode(node, withName: typeIdentifier.name.text, members: node.members.members)
+    }
+    return .visitChildren
+  }
+
+  // MARK: Private
+
+  private let onNodeVisit: (_ info: TypeWithPropInfo) -> Void
+
+  private func processNode<Node>(_ node: Node, withName name: String, members: MemberDeclListSyntax) where Node: SyntaxProtocol {
+    var result = [TypeWithPropInfo.PropInfo]()
+    let propertyVisitor = PropertySyntaxVisitor(parentName: name) { info in
+      result.append(info)
+    }
+    propertyVisitor.walk(node)
+    onNodeVisit(.init(name: name, properties: result))
+  }
+}
+
+// MARK: - PropertySyntaxVisitor
+
+private final class PropertySyntaxVisitor: SyntaxVisitor {
+
+  init(parentName: String, onNodeVisit: @escaping (_ info: TypeWithPropInfo.PropInfo) -> Void) {
+    self.parentName = parentName
+    self.onNodeVisit = onNodeVisit
+  }
+
+  override func visit(_ node: VariableDeclSyntax) -> SyntaxVisitorContinueKind {
+    var leadingTrivia: Trivia? = node.leadingTrivia
+    var access: TypeWithPropInfo.Access = .internal
+    var scope: TypeWithPropInfo.Scope = .instance
+
+    node.modifiers?.forEach { modifier in
+      // leading trivia is on the first modifier, or the node itself if no modifiers are present
+      leadingTrivia = leadingTrivia ?? modifier.leadingTrivia
+      // This attempts to get any access information from the node's modifiers
+      if let modifierAccess = TypeWithPropInfo.Access(rawValue: modifier.name.text) {
+        access = modifierAccess
+      }
+      // this attempts to get any scope information from the node's modifiers
+      if let modifierScope = TypeWithPropInfo.Scope(rawValue: modifier.name.text) {
+        scope = modifierScope
+      }
+    }
+
+    // in practice this `bindings` array only has 1 element
```

Comment:
i don't think this is always true, I think we can have something like:

```swift
struct Foo {
  let some: String, another: Int
}
```

And this will contain 2 bindings IIUC

Reply:
At this point we are already inside of the `VariableDeclSyntax` node. Maybe I just don't understand how the node API works, but wouldn't your example result in 2 separate nodes being parsed instead of 1?

Reply:
I don't think so. That'd be one variable declaration IIUC with two bindings but I haven't validated this. maybe we can add a test for this use case?